flink-yarn/src/main/java/org/apache/flink/yarn/configuration/YarnConfigOptions.java

```diff
@@ -156,7 +155,6 @@ private YarnConfigOptions() {}
 
 	/** @see YarnConfigOptions#CLASSPATH_INCLUDE_USER_JAR */
 	public enum UserJarInclusion {
-		DISABLED,
```

Comment:
We should check where this is parsed to the enum and throw a meaningful exception explaining that DISABLED has been removed.

Reply:
From Flink 1.5 to 1.7:
* If `DISABLED` is set, the job keeps failing.
* If an invalid value is set, the default value (`ORDER`) is used, and a warning is logged.

If we would apply the PR as is, we would always log a warning (because `DISABLED` will be considered an invalid value). You are proposing to throw an exception if `DISABLED` is set. I could do that but I think it makes sense to also propagate the exception for all other invalid values instead of logging a warning. What do you think?

Reply:
+1 for throwing exception rather than just logging, when mismatch.